python/tchannel/tcurl.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import
+
+import argparse
+import logging
+import os
+import sys
+
+import tornado.ioloop
+
+from .tchannel import TChannel
+
+
+log = logging.getLogger('tchannel')
+
+
+@tornado.gen.coroutine
+def tcurl(host, endpoint, headers, body):
+
+    client = TChannel()
+
+    def handle_call(context):
+        """Handle a TChannel CALL_REQ message."""
+        if not context:
+            log.debug("didn't receive context")
+            return
+
+        #connection.pong()
+        log.debug("received message %s", context.message)
+        log.debug(context.message.arg_1)
+        log.debug(context.message.arg_1)
+        log.debug(context.message.arg_2)
+        log.debug(context.message.arg_3)
+
+    print
+    print "Sending this to", host
+    print "*" * 80
+    print " arg1:", endpoint
+    print " arg2:", headers
+    print " arg3:", body
+    print
+
+    request = yield client.request(host)
+
+    response = yield request.send(
```

Comment:
I don't like that there are two yields involved in making this call.

Can `request()` be changed to return a simple object that makes the actual connection on `send()` if required?